packages/ui-view/src/View/types.ts

```diff
@@ -0,0 +1,295 @@
+   * to edges
+   */
+  borderWidth?: BorderWidth
+  /**
+   * Accepts `small`, `medium`, `large`, `circle`, and `pill`. Border radius can be
+   * assigned to individual corners in CSS shorthand style (e.g., `"medium large none pill"`).
+   */
+  borderRadius?: BorderRadii
+  /**
+   * Controls the shadow depth for the `<View />`
+   */
+  shadow?: Shadow
+
+  /**
+   * Controls the z-index depth for the `<View />`
+   */
+  stacking?: Stacking
+  /**
+   * Specify a mouse cursor to use when hovering over the `<View />`
+   */
+  cursor?: any
+}
+
+export type ViewProps = ViewOwnProps &
+  WithStyleProps &
+  OtherHTMLAttributes<ViewOwnProps>
+export type AllowedPropKeys = Readonly<
+  Array<keyof (ViewOwnProps & WithStyleProps)>
+>
+
+export const propTypes = {
+  as: PropTypes.elementType,
+  elementRef: PropTypes.func,
+  display: PropTypes.oneOf([
+    'auto',
+    'inline',
+    'block',
+    'inline-block',
+    'flex',
+    'inline-flex'
+  ]),
+  overflowX: PropTypes.oneOf(['auto', 'hidden', 'visible']),
+  overflowY: PropTypes.oneOf(['auto', 'hidden', 'visible']),
+  margin: ThemeablePropTypes.spacing,
+  padding: ThemeablePropTypes.spacing,
+  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  maxHeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  maxWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  minHeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  minWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  children: PropTypes.node,
+  textAlign: PropTypes.oneOf(['start', 'center', 'end']),
+  borderWidth: ThemeablePropTypes.borderWidth,
+  borderRadius: ThemeablePropTypes.borderRadius,
+  borderColor: PropTypes.oneOf([
+    'transparent',
+    'primary',
+    'secondary',
+    'brand',
+    'info',
+    'success',
+    'warning',
+    'alert',
+    'danger'
+  ]),
+  background: PropTypes.oneOf([
+    'transparent',
+    'primary',
+    'secondary',
+    'primary-inverse',
+    'brand',
+    'info',
+    'alert',
+    'success',
+    'danger',
+    'warning'
+  ]),
+  shadow: ThemeablePropTypes.shadow,
+  stacking: ThemeablePropTypes.stacking,
+  cursor: cursorPropTypes,
+  position: PropTypes.oneOf([
+    'static',
+    'absolute',
+    'relative',
+    'sticky',
+    'fixed'
+  ]),
+  insetInlineStart: PropTypes.string,
+  insetInlineEnd: PropTypes.string,
+  insetBlockStart: PropTypes.string,
+  insetBlockEnd: PropTypes.string,
+  withFocusOutline: PropTypes.bool,
+  focusPosition: PropTypes.oneOf(['offset', 'inset']),
+  focusColor: PropTypes.oneOf(['info', 'inverse', 'success', 'danger']),
+  shouldAnimateFocus: PropTypes.bool,
+  withVisualDebug: PropTypes.bool,
+  makeStyles: PropTypes.func,
+  styles: PropTypes.object,
+  dir: PropTypes.oneOf(Object.values(bidirectional.DIRECTION))
+}
+
+export const allowedProps: AllowedPropKeys = [
```

Comment:
I would add some comments above it about what it and why we use it like that

Reply:
I've added some explanation comments, could you check it again please?

Reply:
Thanks!